homeassistant/components/mazda/__init__.py

```diff
@@ -52,7 +53,9 @@ async def with_timeout(task, timeout_seconds=10):
         return await task
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(  # noqa: C901
```

Comment:
Why the noqa?

Reply:
Because of the deprecation warning I added, there is now a linter warning for "function is too complex". I guess that one additional branch pushed it just over the edge. I figured it was fine to suppress the linter warning, since the deprecation warnings will be removed in a couple of releases, at which time I will remove the noqa. I also see this particular linter warning being suppressed in many other places throughout the codebase. If this is not acceptable then please let me know and I can try to refactor the function to avoid the warning.

Reply:
You could move the code in embedded functions out into a new function and call them from the existing function so you still are holding a reference to hass

Reply:
Ok, but is this really needed for this PR? It seems a bit out of scope.

Can we leave the noqa there since it will be removed in a couple of releases? The code in `async_handle_service_call()` will also become simpler at that time, since all services except `send_poi` will be removed.

Reply:
It would be better to refactor it in a fresh PR before merging this one. I prefer not to merge code that turns off the checks unless its for a bug fix since there is no way to ensure that they get turned back on in the future.

Reply:
I was able to remove the noqa by making a small change to the function.